libcudacxx/test/libcudacxx/std/strings/c.strings/cstring.syn/string_exam/strchr.pass.cpp

```diff
@@ -0,0 +1,84 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#include <cuda/std/cassert>
+#include <cuda/std/cstddef>
+#include <cuda/std/cstring>
+#include <cuda/std/limits>
+#include <cuda/std/type_traits>
+
+#include "test_macros.h"
+
+__host__ __device__ constexpr void test_strchr(char* str, int c, char* expected_ret)
+{
+  const char* cstr = const_cast<const char*>(str);
+
+  // Test cuda::std::strchr(char*, int) overload
+  {
+    const auto ret = cuda::std::strchr(str, c);
+    assert(ret == expected_ret);
+  }
+
+  // Test cuda::std::strchr(const char*, int) overload
+  {
+    const auto ret = cuda::std::strchr(cstr, c);
+    assert(ret == expected_ret);
+  }
+}
+
+__host__ __device__ constexpr bool test()
+{
+  static_assert(cuda::std::is_same_v<char*, decltype(cuda::std::strchr((char*) {}, int{}))>);
```

Comment:
I believe this should rather be
```suggestion
  static_assert(cuda::std::is_same_v<char*, decltype(cuda::std::strchr(static_cast<char*>(nullptr), 0))>);
```
Applies throughout

Reply:
I used `declval` to fix it